stdlib/public/core/Collection.swift

```diff
@@ -1044,6 +1044,11 @@ extension Collection where SubSequence == Slice<Self> {
   }
 }
 
+extension Collection {
+  @available(*, unavailable)
```

Comment:
Good point.

Reply:
We should also add an `@_alwaysEmitIntoClient` attribute so that we don't create an ABI entry point that could become a liability later. 

(Without aeic, we would technically also need to describe which OS versions introduced this entry point.)

Reply:
To emphasize this: Swift code is allowed to link to unavailable symbols -- e.g., IIRC, without aeic, the code below would build fine and it would get compiled into a binary that has strong linkage to the unavailable slicing subscript:

```swift
@available(*, unavailable)
struct CollectionWithNonDefaultSubSequence: Collection {
   public var startIndex: Int
   public var endIndex: Int

   public typealias SubSequence = Self

   public func index(after i: Int) -> Int { i+1 }
   public subscript(position: Int) -> Int { position }
 }
```

Unavailable code is still compiled and emitted into the binary; it just usually isn't expected to ever run. (Except when it is, e.g. in the case of some entry points in dylibs that support running on both vanilla macOS and Mac Catalyst.)

Reply:
Should `@_alwaysEmitIntoClient` also be added to the RNG fix, then?

Reply:
Oh yes! And we probably want to merge that on 5.5 too.